lib/common/common/src/panic.rs

```diff
@@ -0,0 +1,15 @@
+use std::any::Any;
+
+type PanicPayload = Box<dyn Any + Send + 'static>;
+
+/// Convert a panic payload into a string
+///
+/// This converts `String` and `&str` panic payloads into a string.
+/// Other payload types are formatted as is, and may be non descriptive.
+pub fn panic_payload_into_string(payload: PanicPayload) -> String {
+    payload
+        .downcast::<&str>()
+        .map(|msg| msg.to_string())
+        .or_else(|payload| payload.downcast::<String>().map(|msg| msg.to_string()))
+        .unwrap_or_else(|payload| format!("{payload:?}"))
+}
```

Comment:
I'd use my `downcast_str`:
- does not require `Send` and `'static`
- does not consume `payload`
- does not allocate
- also, IMO, it is easier to read the way I structured it :D

Reply:
Sure! I don't really mind since we only call this on panic.

I selected the owned version because you cannot `format!("{payload:?}")` into a `&str`.

Reply:
Does it? Wasn't aware of that.

In that case sorry for destroying your work :clown_face:

Reply:
Fixed.